### ISSUE TITLE
Support html specific serialization

### ISSUE THREAD

**Issue by skabbes** [polotek/libxmljs#213]
When parsing and serializing an html document, you can get invalid html if you don't use the correct html "toString" provided by libxml2

Parsing and serializing this html
```html
<html>
  <head>
  <title></title>
  </head>
  <body>
  Hello world
  </body>
</html>
```

Will result in (note the empty title tag)
```html
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html>
  <head>
    <title/>
  </head>
  <body>Hello World</body>
</html>
```

Which will render incorrectly in most web browsers due to [this bug](https://code.google.com/p/html5lib/issues/detail?id=195)

However, if we render it with [htmlDocDumpMemoryFormat](http://xmlsoft.org/html/libxml-HTMLtree.html#htmlDocDumpMemoryFormat) (and with [htmlParserOption: 4](http://www.xmlsoft.org/html/libxml-HTMLparser.html#htmlParserOption) to ignore the implicit doctype declaration) we get this instead.
```html
<html>
<head><title></title></head>
<body>Hello World</body>
</html>
```

The new method will actually render correctly in all browsers.

I've [fixed this](https://github.com/CodeFridge/libxmljs_mem/commit/d1f360406aeb13374dedd361904587dc81388d2d) in my fork.  However I'd like to get input on how you might do this more generically (I see the HtmlDocument class referenced, but unused).

**Comment by neu-rah** [comment]
yes, please

**Comment by rc0x03** [comment]
:+1:

**Comment by defunctzombie** [comment]
we can make toString accept options arguments and pick the right function based on that

**Comment by teleological** [comment]
Implemented in PR #326 / #341

**Comment by rc0x03** [comment]
`XmlDocument::toString` now checks for HTML and uses `htmlDocDumpMemoryFormat`. This issue has been resolved.
